### PR TITLE
Withdraw escrow

### PR DESCRIPTION
**Pull request by dwalintukan**
image

### CODE REVIEW

File: src/redux/Topic/saga.js [hunk 1]
```diff
@@ -116,19 +146,24 @@ export function* calculateWinningsHandler() {
...
         // return only winning addresses
         if (botWon || qtumWon) {
-          value.push({ address: senderAddress, botWon, qtumWon });
+          withdrawableAddresses.push({
+            type: WithdrawType.winnings,
+            address: vote.voterQAddress,
+            botWon,
+            qtumWon,
+          });
         }
```
Comment (liviavinci) [inline src/redux/Topic/saga.js]: missing the escrow address i believe?

Comment (dwalintukan) [reply src/redux/Topic/saga.js]: it's not needed. the escrow holder is the AddressManager. As long as the BE knows the topicAddress, the topic has the AddressManager's address so it knows where to withdraw from.

File: src/network/graphMutation.js [hunk 2]
```diff
@@ -129,8 +129,9 @@ export function createFinalizeResultTx(version, topicAddress, oracleAddress, sen
   return new GraphMutation('finalizeResult', args, TYPE.transaction).execute();
 }
 
-export function createWithdrawTx(version, topicAddress, senderAddress) {
+export function createWithdrawTx(type, version, topicAddress, senderAddress) {
   const args = {
+    type,
```
Comment (alex-cory) [inline src/network/graphMutation.js]: you know you can get all the args like this?
```
export const createWithdrawTx = (...args) => new GraphMutation('withdraw', args, TYPE.transaction).execute();
```

Comment (alex-cory) [reply src/network/graphMutation.js]: jk, they're an array instead of an object then.

Comment (dwalintukan) [reply src/network/graphMutation.js]: lol

File: src/scenes/Event/scenes/topic.js [hunk 3]
```diff
@@ -360,31 +345,32 @@ export default class TopicPage extends React.Component {
...
-    const config = this.getActionButtonConfig(walletWithinWinnings.address);
+    const config = this.getActionButtonConfig(withdrawableAddress.address);
 
     if (!config.show) {
       return null;
     }
 
     return (
-      <TableRow key={walletWithinWinnings.address}>
+      <TableRow key={index}>
```
Comment (liviavinci) [inline src/scenes/Event/scenes/topic.js]: Doesn't using index as key give you a warning?

File: src/scenes/Event/scenes/topic.js [hunk 4]
```diff
@@ -360,31 +345,32 @@ export default class TopicPage extends React.Component {
...
         <TableCell padding="dense">
-          <div>{walletWithinWinnings.address}</div>
+          <div>{withdrawableAddress.address}</div>
           <div className={config.warningTypeClass}>{config.message}</div>
         </TableCell>
-        <TableCell padding="dense">Winning</TableCell>
+        <TableCell padding="dense">{withdrawableAddress.type.toUpperCase()}</TableCell>
```
Comment (liviavinci) [inline src/scenes/Event/scenes/topic.js]: localization?

File: src/scenes/Event/scenes/topic.js (hunk 3 above)
Comment (dwalintukan) [reply src/scenes/Event/scenes/topic.js]: nope. it was actually giving me an error if i had an escrow and winnings withdraw. would only show one.

File: src/scenes/Event/scenes/topic.js (hunk 4 above)
Comment (dwalintukan) [reply src/scenes/Event/scenes/topic.js]: done

**Pull request merged**